kola/tests/devcontainer/devcontainer.go

```diff
@@ -225,18 +228,32 @@ func init() {
 		Distros:   []string{"cl"},
 		// TODO: Revisit this flag when updating SELinux policies.
 		Flags: []register.Flag{register.NoEnableSelinux},
+		NativeFuncs: map[string]func() error{
+			"Http": Serve,
+		},
 	})
 }
 
 func withSystemdNspawn(c cluster.TestCluster) {
-	runDevContainerTest(c, defaultScriptTemplateParameters, systemdNspawnScriptBody)
+	runDevContainerTest(c, systemdNspawnScriptBody)
 }
 
 func withDocker(c cluster.TestCluster) {
-	runDevContainerTest(c, defaultScriptTemplateParameters, dockerScriptBody)
+	runDevContainerTest(c, dockerScriptBody)
 }
 
-func runDevContainerTest(c cluster.TestCluster, scriptParameters scriptTemplateParameters, scriptBody string) {
+func runDevContainerTest(c cluster.TestCluster, scriptBody string) {
+	devcontainerURL := kola.DevcontainerURL
+	if kola.DevcontainerFile != "" {
+		// This URL is deterministic as it runs on the started machine.
+		devcontainerURL = "http://localhost:8080"
+	}
+
+	scriptParameters := scriptTemplateParameters{
+		BinhostURLTemplate:        "http://bincache.flatcar-linux.net/boards/@ARCH@-usr/@VERSION@/pkgs",
```

Comment:
Do we also need to pass the `pkgs` folder to the instance? This bincache path only works for developer builds made on Jenkins but since this here is meant to be used in GitHub Actions, nothing will be there.

Reply:
Oh right, I did not even consider this because I used images from a nightly build for my testing above.
We would also need to export the packages in the Build CI job and fetch that artifact in the kola CI job. Making a metal note for my work on https://github.com/flatcar/Flatcar/issues/794.

Reply:
So, the idea would be to run a local `python -m http.server` before running kola to give it access to the pkgs folder and then use the new image flag for the image?

Reply:
Or rather not use the new image flag because it would occupy disk space on the VM, and use the temporary python http server for both.

Reply:
Since this runs in VMs with private addresses I guess it will be enough to let it bind to 0.0.0.0 as done by default and then use the VM's internal IP address for the HTTP URL